R/derive_param_clinbenefit.R

```diff
@@ -204,9 +219,12 @@ derive_param_clinbenefit <- function(dataset,
     enquo(filter_source),
     optional = FALSE
   )
-  assert_s3_class(source_resp, "date_source", optional = FALSE)
-  assert_s3_class(source_pd, "date_source", optional = TRUE)
-  assert_list_of(source_datasets, "data.frame", optional = FALSE)
+
+  assert_function(aval_fun)
+  assert_s3_class(source_resp, "date_source")
```

Comment:
the default for the optional argument in assert_s3_class and assert_list_of is TRUE, so is it correct to not have the argument set to FALSE here when checking source_resp and source_datasets? This is why I set the argument optional == FALSE (and for clarity set the source_pd to be TRUE) when I made the recent changes for source_pd

Reply:
@bundfussr what are your thoughts for this?

Reply:
In general, I would not set arguments to its default value in a call. If we there are reasons to set an argument even if it is the default, I would prefer removing the default value.

You are right that we should set `optional = FALSE` for `assert_s3_class()` and `assert_list_of()`. I am not sure why the default is `TRUE` for these functions. I would prefer to unify it, i.e., for all assertions the default should be `FALSE`. I'll set it on the agenda of the next admiral core meeting.

Reply:
https://github.com/pharmaverse/admiralonco/issues/151